PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/core/FunctionUtils.java

```diff
@@ -161,8 +175,11 @@ public static void prepareStagingFolder(Path stagingFolder, Path hostJson, Path
             }
         }
 
-        copyFilesWithDefaultContent(hostJson, new File(stagingFolder.toFile(), "host.json"), DEFAULT_HOST_JSON);
-        copyFilesWithDefaultContent(localSettingJson, new File(stagingFolder.toFile(), "local.settings.json"), DEFAULT_LOCAL_SETTINGS_JSON);
+        final File hostJsonFile = new File(stagingFolder.toFile(), "host.json");
+        copyFilesWithDefaultContent(hostJson, hostJsonFile, DEFAULT_HOST_JSON);
+        final File localSettingsFile = new File(stagingFolder.toFile(), "local.settings.json");
+        copyFilesWithDefaultContent(localSettingJson, localSettingsFile, DEFAULT_LOCAL_SETTINGS_JSON);
+        updateLocalSettingValues(localSettingsFile, appSettings);
```

Comment:
need to check when appSettings is null or empty, to avoid write two times.

Reply:
Done